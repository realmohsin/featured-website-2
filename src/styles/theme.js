import { createMuiTheme } from '@material-ui/core/styles'

const primaryColor = '#0E5C60'
const primaryDarkColor = '#002B2E'
const primaryVariant1 = '#00464A'
const secondaryColor = '#CCAB63'
const goldVariant1 = '#E1C079'
const facebookColor = '#3C5B9B'
const googleStarColor = '#E7711B'
const fadedTextColor = '#94B7B8'

const theme = createMuiTheme({
  palette: {
    primary: {
      main: primaryColor,
      dark: primaryDarkColor
    },
    secondary: {
      main: secondaryColor
    },
    common: {
      primaryVariant1,
      goldVariant1,
      facebookColor,
      googleStarColor,
      fadedTextColor
    }
  },
  typography: {
    htmlFontSize: 10,
    fontFamily: ['Roboto', 'sans-serif'].join(', '),
    body1: {
      fontSize: '1.6rem',
      '@media (max-width: 600px)': {
        fontSize: '1.4rem'
      }
    },
    body2: {
      fontSize: '1.4rem',
      lineHeight: '1.5'
    }
  },
  overrides: {
    MuiCssBaseline: {
      '@global': {
        '*, *::before, *::after': {
          margin: 0,
          padding: 0
        },
        html: {
          fontSize: '62.5%'
        },
        a: {
          display: 'block',
          textDecoration: 'none',
          color: 'inherit'
        },
        img: {
          display: 'block'
        },
        'input, button, textarea, select': {
          display: 'block',
          font: 'inherit'
        },
        ul: {
          listStyle: 'none'
        },
        '::selection': {
          backgroundColor: primaryColor,
          color: 'white'
        },
        'a, button': {
          cursor: 'pointer'
        }
      }
    },
    MuiTypography: {
      gutterBottom: {
        marginBottom: '1.2em'
      }
    }
  },
  custom: {
    title1: {
      fontFamily: ['Piazzolla', 'serif'].join(', '),
      fontSize: '5.1rem',
      color: primaryColor,
      '@media (max-width: 600px)': {
        fontSize: '2rem'
      }
    },
    title2: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '4.6rem',
      color: primaryColor,
      fontWeight: 'normal',
      marginBottom: '8rem',
      textAlign: 'center'
    },
    title3: {
      fontFamily: ['PT Serif', 'serif'].join(', '),
      fontSize: '3.4rem',
      color: secondaryColor,
      fontWeight: 'normal'
    },
    title4: {
      fontSize: '2.2rem',
      color: primaryColor,
      textTransform: 'uppercase',
      wordSpacing: '0.5px'
    },
    title5: {
      fontSize: '1.5rem',
      color: secondaryColor,
      textTransform: 'uppercase',
      fontWeight: 'normal'
    },
    outlineTitle: {
      textShadow: `-1px 0 ${primaryColor}, 0 1px ${primaryColor}, 1px 0 ${primaryColor}, 0 -1px ${primaryColor}`
    },
    inSideDrawer: {
      transform: 'translateY(0)',
      opacity: 1,
      pointerEvents: 'auto',
      width: 'auto',
      height: 'auto',
      border: 'none',
      backgroundColor: 'white',
      boxShadow: 'inset 0 0 5px 2px rgba(0,0,0,.2)',
      '& a': {
        color: primaryColor,
        fontSize: '1.1rem',
        display: 'flex',
        height: '4rem',
        borderBottom: `1px solid #eee`,
        alignItems: 'center',
        justifyContent: 'space-between',
        paddingRight: '1rem',
        fontWeight: 'normal',
        paddingLeft: '3.6rem'
      }
    }
  }
})

export default theme
