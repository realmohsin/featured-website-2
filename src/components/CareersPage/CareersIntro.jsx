import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import { Container, Typography } from '@material-ui/core'

const useStyles = makeStyles(theme => ({
  careersIntroSection: {
    padding: '5rem 0 0',
    [theme.breakpoints.down('xs')]: {
      padding: '2rem 0.6rem 0'
    }
  },
  careersTitle: {
    ...theme.custom.title2,
    // textAlign: 'left',
    marginTop: '2rem',
    marginBottom: '3rem',
    fontSize: '5.6rem',
    [theme.breakpoints.down('xs')]: {
      fontSize: '3.4rem'
    }
  },
  joinSubtitle: {
    ...theme.custom.title3,
    marginBottom: '3rem',
    [theme.breakpoints.down('xs')]: {
      fontSize: '2.2rem',
      textAlign: 'center'
    }
    // textShadow: `-1px 0 ${theme.palette.primary.main}, 0 1px ${theme.palette.primary.main}, 1px 0 ${theme.palette.primary.main}, 0 -1px ${theme.palette.primary.main}`
  },
  positionLink: {
    marginBottom: '1.5rem',
    textDecoration: 'underline',
    fontSize: '1.5rem',
    background: 'none',
    border: 'none',
    color: 'inherit',
    cursor: 'pointer',
    padding: 0,
    display: 'block',
    textAlign: 'left',
    '&:hover': {
      opacity: 0.8
    }
  }
}))

const CareersIntro = props => {
  const classes = useStyles()

  return (
    <section className={classes.careersIntroSection}>
      <Container maxWidth='md'>
        <Typography variant='h1' className={classes.careersTitle}>
          Careers
        </Typography>
        <Typography variant='h2' className={classes.joinSubtitle}>
          Join Our Team
        </Typography>
        <div>
          <Typography variant='body1' gutterBottom>
            We are Innovative Problem Solvers.
          </Typography>
          <Typography variant='body1' gutterBottom>
            Romano Law is a boutique law firm of talented and diverse people.
            Located in Downtown Manhattan, we are focused on Business Law,
            Entertainment, Employment and Dispute Resolution. Whatever
            “traditional” thoughts you might have about a law firm, that’s not
            us. We do things differently. Our team-oriented culture is
            professional, but modern – always working on growing and improving.{' '}
          </Typography>
          <Typography variant='body1' gutterBottom>
            We are all about getting things done and having fun along the way.
            We’re looking for intelligent, dynamic and thoughtful people.
          </Typography>
          <Typography variant='body1' gutterBottom>
            The following positions are available:
          </Typography>
          <button type="button" className={classes.positionLink}>
            Office Manager
          </button>
          <button type="button" className={classes.positionLink}>
            Experienced Corporate Attorney
          </button>
          <button type="button" className={classes.positionLink}>
            Client Development Director
          </button>
          <button type="button" className={classes.positionLink}>
            Virtual Assistant
          </button>
          <button type="button" className={classes.positionLink}>
            Experienced Business Litigation Attorney
          </button>
        </div>
      </Container>
    </section>
  )
}

export default CareersIntro
