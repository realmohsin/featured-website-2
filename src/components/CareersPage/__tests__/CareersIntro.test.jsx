import React from "react";
import { render, screen } from "@testing-library/react";
import { ThemeProvider } from "@material-ui/core/styles";
import { createMuiTheme } from "@material-ui/core/styles";
import CareersIntro from "../CareersIntro";

const theme = createMuiTheme({
  custom: {
    title2: {
      fontSize: "5.6rem",
      fontWeight: "bold",
    },
    title3: {
      fontSize: "3.2rem",
    },
  },
  breakpoints: {
    down: () => "@media (max-width: 600px)",
  },
});

const renderWithTheme = (component) => {
  return render(<ThemeProvider theme={theme}>{component}</ThemeProvider>);
};

describe("CareersIntro Component", () => {
  it("renders careers intro section", () => {
    const { container } = renderWithTheme(<CareersIntro />);
    expect(container.firstChild).toMatchSnapshot();
  });

  it("displays main headings", () => {
    renderWithTheme(<CareersIntro />);
    expect(screen.getByRole("heading", { name: /careers/i })).toBeInTheDocument();
    expect(screen.getByRole("heading", { name: /join our team/i })).toBeInTheDocument();
  });

  it("displays company description text", () => {
    renderWithTheme(<CareersIntro />);
    expect(screen.getByText(/we are innovative problem solvers/i)).toBeInTheDocument();
    expect(screen.getByText(/romano law is a boutique law firm/i)).toBeInTheDocument();
    expect(screen.getByText(/we are all about getting things done/i)).toBeInTheDocument();
    expect(screen.getByText(/the following positions are available/i)).toBeInTheDocument();
  });

  it("displays all available positions", () => {
    renderWithTheme(<CareersIntro />);
    
    expect(screen.getByRole("button", { name: /office manager/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /experienced corporate attorney/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /client development director/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /virtual assistant/i })).toBeInTheDocument();
    expect(screen.getByRole("button", { name: /experienced business litigation attorney/i })).toBeInTheDocument();
  });

  it("renders position buttons with correct styling", () => {
    renderWithTheme(<CareersIntro />);
    
    const positionButtons = screen.getAllByRole("button");
    expect(positionButtons).toHaveLength(5);
    
    positionButtons.forEach(button => {
      expect(button).toHaveAttribute("type", "button");
    });
  });
});
