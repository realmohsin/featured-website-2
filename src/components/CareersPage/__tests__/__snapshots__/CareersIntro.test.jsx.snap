// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CareersIntro Component renders careers intro section 1`] = `
<section
  class="makeStyles-careersIntroSection-1"
>
  <div
    class="MuiContainer-root MuiContainer-maxWidthMd"
  >
    <h1
      class="MuiTypography-root makeStyles-careersTitle-2 MuiTypography-h1"
    >
      Careers
    </h1>
    <h2
      class="MuiTypography-root makeStyles-joinSubtitle-3 MuiTypography-h2"
    >
      Join Our Team
    </h2>
    <div>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        We are Innovative Problem Solvers.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        Romano Law is a boutique law firm of talented and diverse people. Located in Downtown Manhattan, we are focused on Business Law, Entertainment, Employment and Dispute Resolution. Whatever “traditional” thoughts you might have about a law firm, that’s not us. We do things differently. Our team-oriented culture is professional, but modern – always working on growing and improving.
         
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        We are all about getting things done and having fun along the way. We’re looking for intelligent, dynamic and thoughtful people.
      </p>
      <p
        class="MuiTypography-root MuiTypography-body1 MuiTypography-gutterBottom"
      >
        The following positions are available:
      </p>
      <button
        class="makeStyles-positionLink-4"
        type="button"
      >
        Office Manager
      </button>
      <button
        class="makeStyles-positionLink-4"
        type="button"
      >
        Experienced Corporate Attorney
      </button>
      <button
        class="makeStyles-positionLink-4"
        type="button"
      >
        Client Development Director
      </button>
      <button
        class="makeStyles-positionLink-4"
        type="button"
      >
        Virtual Assistant
      </button>
      <button
        class="makeStyles-positionLink-4"
        type="button"
      >
        Experienced Business Litigation Attorney
      </button>
    </div>
  </div>
</section>
`;
